### ISSUE TITLE
Server refuses key

### ISSUE THREAD

**Issue by X00LA** [rmichela/Bukkit-SSHD#4]
Hi, I would use your plugin but when I want to connect I always get the message that my server refuses the key.

**Comment by Haarolean** [comment]
Hi there, are you sure you're using proper keys format?

**Comment by X00LA** [comment]
Yes, I created the keys like you mentioned in your thread.
Copied the key into authorized_keys and set chmod to 744.
Then added the private key to my putty and tried to connect.

It's normal for me to run all SSHD connections over key files and didn't have any problems with it.
Could it be possible that this causes the problem? I use SSHD key based auth to connect my minecraft user to manage my files. I tried it with the key file from my existing user and with a key file I have created for another user. But both connections get refused. With my normal user I'm able to connect without any problem.

**Comment by Haarolean** [comment]
Are you totally sure you're not using PuTTY' key format on server side?

**Comment by X00LA** [comment]
100%
I created the key files step by step as you described in your thread on spigot. Tried both methods, Win and Linux (Ubuntu 16.04). No one works.

**Comment by Haarolean** [comment]
Can you generate a test keypair and share it? I'll test on my side

**Comment by X00LA** [comment]
Here they are.
I have insert the generated id_rsa, id_rsa.pub, the file with the custom name and a text file with the passphrase..

[SSH-Keys.zip](https://github.com/rmichela/Bukkit-SSHD/files/1676220/SSH-Keys.zip)

**Comment by Haarolean** [comment]
Yeah, here's the problem. You need to convert your key in PEM format, which is explained in p. 3 on bukkitdev/spigot.
![image](https://user-images.githubusercontent.com/1494347/35562067-4b6df288-05c3-11e8-8485-e6940fe51e24.png)
Your public key should start like this:

>
>---- BEGIN SSH2 PUBLIC KEY ----
>Comment: "2048-bit RSA, converted by haarolean@haarolean-pc from OpenS"
>

**Comment by Haarolean** [comment]
Your key works for me fine. Try your key with ssh testtest@cronfire.ru -p 45005 -i <yourkey>

**Comment by X00LA** [comment]
Sorry, can't test it. My wife has deleted the files because she tought this is malware. :/ (Haven't saved it in my personal folder.)
Could you please post the archive to download it.

Thank you.

**Comment by Haarolean** [comment]
Sure.
[SSH-Keys.zip](https://github.com/rmichela/Bukkit-SSHD/files/1677429/SSH-Keys.zip)

**Comment by X00LA** [comment]
Thank you!

I will test it now.

**Comment by X00LA** [comment]
Hmmm...
Why does'nt it work on my server?
I have all set up right. My other connections work fine.
I have no clou whats wrong.

**Comment by Haarolean** [comment]
Can you show the filetree under plugins/SSHD?

**Comment by X00LA** [comment]
MCServer -- plugins -- SSHD -- config.yml
MCServer -- plugins -- SSHD -- authorized_keys -- xoola

**Comment by Haarolean** [comment]
The xoola file should be without file extension and in PEM format. Have you tried different chmod? Also, any messages in log?

**Comment by X00LA** [comment]
It's the same file I send you. Tried multiple chmod. No file extension, no messages in the log.
I will trie it on my other server. When it will work there, I have anything setup wrong here.

I send you a message when its done.

**Comment by X00LA** [comment]
Finally I have fixed it!
I tested it on my second server and it runs fine.
So why the hell it does'nt work on my main server?

I changed the settings in sshd_config, nothing happend.
I changed firewall rules, nothing happend.
I changed permissions, nothing happend.

Ok, I dont give up!

I created a new user and installed the server within the new users home folder.
I cross fingers and fire up my minecraft server...

Ok, I try to connect...

IT WORKS! I cant believe!

No clou what happens with the account where I was running minecraft before, but now it works fine.

Sorry for the problems I've maked.

THANK YOU FOR THE HELP!
